src/main/java/com/esri/samples/na/closest_facility_static/ClosestFacilityStaticSample.java

```diff
@@ -0,0 +1,319 @@
+              }
+
+            } catch (InterruptedException | ExecutionException e) {
+              displayMessage("Error retrieving list of incidents", e.getMessage());
+            }
+          });
+
+          // enable the 'solve routes' button
+          solveRoutesButton.setDisable(false);
+
+          // resolve button press
+          solveRoutesButton.setOnAction(e -> {
+
+            // disable the 'solve routes' button and show the progress indicator
+            solveRoutesButton.setDisable(true);
+            progressIndicator.setVisible(true);
+
+            // start the routing task
+            closestFacilityTask.loadAsync();
+            closestFacilityTask.addDoneLoadingListener(() -> {
+              if (closestFacilityTask.getLoadStatus() == LoadStatus.LOADED) {
+                try {
+                  // create default parameters for the task and add facilities and incidents to parameters
+                  ClosestFacilityParameters closestFacilityParameters = closestFacilityTask.createDefaultParametersAsync().get();
+                  closestFacilityParameters.setFacilities(facilities);
+                  closestFacilityParameters.setIncidents(incidents);
+
+                  // solve closest facilities
+                  try {
+                    // use the task to solve for the closest facility
+                    ListenableFuture<ClosestFacilityResult> closestFacilityTaskResult = closestFacilityTask.solveClosestFacilityAsync(closestFacilityParameters);
+                    closestFacilityTaskResult.addDoneListener(() -> {
+                      try {
+                        ClosestFacilityResult closestFacilityResult = closestFacilityTaskResult.get();
+
+                        // find the closest facility for each incident
+                        for (int incidentIndex = 0; incidentIndex < incidents.size(); incidentIndex++) {
+
+                          // get the index of the closest facility to incident
+                          Integer closestFacilityIndex = closestFacilityResult.getRankedFacilityIndexes(incidentIndex).get(0);
+
+                          // get the route to the closest facility
+                          ClosestFacilityRoute closestFacilityRoute = closestFacilityResult.getRoute(closestFacilityIndex, incidentIndex);
+
+                          // display the route on the graphics overlay
+                          graphicsOverlay.getGraphics().add(new Graphic(closestFacilityRoute.getRouteGeometry(), simpleLineSymbol));
+
+                          // hide the progress indicator and enable the reset button
+                          progressIndicator.setVisible(false);
+                          resetButton.setDisable(false);
+                        }
+
+                      } catch (ExecutionException | InterruptedException ex) {
+                        displayMessage("Error getting the ClosestFacilityTask result", ex.getMessage());
```

Comment:
Instead of `displayMessage` like we have in older samples, you could do

```java
new Alert(Alert.AlertType.ERROR, "Error getting the closest facility task result").show();
```

The user wouldn't see the stack trace, but in general you would never want to show a stack trace in an alert to an end user anyway.

Reply:
Thanks, I've noticed the two different ways samples show errors.
It sounds like the preference is now to do `new Alert(...)`?
What's the effective difference between the two ways, I can't quite tell what the `displayMessage` method is trying to achieve differently?